crates/lib/src/jekyll/array.rs

```diff
@@ -10,6 +13,122 @@ use liquid_core::{Value, ValueView};
 
 use crate::invalid_input;
 
+#[derive(Debug, Default, FilterParameters)]
+struct SortArgs {
+    #[parameter(description = "The property accessed by the filter.", arg_type = "str")]
+    property: Option<Expression>,
+    #[parameter(
+        description = "nils appear before or after non-nil values, either ('first' | 'last')",
+        arg_type = "str"
+    )]
+    nils: Option<Expression>,
+}
+
+#[derive(Clone, ParseFilter, FilterReflection)]
+#[filter(
+    name = "sort",
+    description = "Sorts items in an array. The order of the sorted array is case-sensitive.",
+    parameters(SortArgs),
+    parsed(SortFilter)
+)]
+pub struct Sort;
+
+#[derive(Debug, Default, FromFilterParameters, Display_filter)]
+#[name = "sort"]
+struct SortFilter {
+    #[parameters]
+    args: SortArgs,
+}
+
+enum NilsOrder {
+    First,
+    Last,
+}
+
+fn safe_property_getter<'a>(value: &'a Value, property: String) -> Value {
+    let properties = property.split('.').collect::<Vec<&str>>();
```

Comment:
Is there a better way to do this? This won't work when the `object["property"]` syntax is used instead of `object.property`, right?